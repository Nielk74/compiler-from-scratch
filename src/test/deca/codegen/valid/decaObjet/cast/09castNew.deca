// Description:
//    Test de cast pour l'expression new
//
// Resultats:
//      09castNew.expected
//
// Historique:
//    cree le 21/06/2022

class A {
    int x=1;
    void m(){
        println("This is from A");
    }
}
class B extends A {
    int x=2;
    void m(){
        println("This is from B");
    }
}
class C extends B {
    int x=3;
    void m(){
        println("This is from C");
    }
}

{
    A c = (B)(new C());
    c.m();
    println(c.x);
    println(((A)(new B())).x);
}