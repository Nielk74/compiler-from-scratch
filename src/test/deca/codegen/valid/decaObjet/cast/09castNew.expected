This is from C
1
1
